xknx/io/self_description.py

```diff
@@ -0,0 +1,80 @@
+"""Abstraction to send DescriptionRequest and wait for DescriptionResponse."""
+from __future__ import annotations
+
+import asyncio
+import logging
+from typing import TYPE_CHECKING, Final
+
+from xknx.exceptions import XKNXException
+from xknx.io.gateway_scanner import GatewayDescriptor
+from xknx.knxip import HPAI, DescriptionRequest, DescriptionResponse, KNXIPFrame
+
+if TYPE_CHECKING:
+    from xknx.io.transport import KNXIPTransport
+    from xknx.xknx import XKNX
+
+logger = logging.getLogger("xknx.log")
+
+DESCRIPTION_TIMEOUT: Final = 2
+
+
+class RequestDescription:
```

Comment:
Should we rename this to something different? This can lead to a lot of confusion as there also is a DescriptionRequest - maybe something like DescriptionRequestExecutor or similar would suit better?

Reply:
What do you think about `DescriptionQuery`?

Reply:
Fine for me